record.c

```diff
@@ -255,6 +256,31 @@ janus_recorder *janus_recorder_create_full(const char *dir, const char *codec, c
 	return rc;
 }
 
+int janus_recorder_pause(janus_recorder *recorder) {
+	if(!recorder)
+		return -1;
+	g_atomic_int_set(&recorder->paused, 1);
+	return 0;
+}
+
+int janus_recorder_resume(janus_recorder *recorder) {
+	if(!recorder)
+		return -1;
+	janus_mutex_lock_nodebug(&recorder->mutex);
+	if(recorder->type == JANUS_RECORDER_AUDIO) {
+		recorder->context.a_ts_reset = TRUE;
+		recorder->context.a_seq_reset = TRUE;
+		recorder->context.a_last_time = janus_get_monotonic_time();
+	} else if(recorder->type == JANUS_RECORDER_VIDEO) {
+		recorder->context.v_ts_reset = TRUE;
+		recorder->context.v_seq_reset = TRUE;
+		recorder->context.v_last_time = janus_get_monotonic_time();
+	}
+	g_atomic_int_set(&recorder->paused, 0);
+	janus_mutex_unlock_nodebug(&recorder->mutex);
```

Comment:
See discussion above: maybe, besides checks at the plugin level, this code should check the status of the recorder too, and not reset these properties if the recorder is not paused already.